### PR TITLE
Improve code coverage on FHIR Core

### LINKED ISSUE #326: Improve code coverage on FHIR Core
We need to increase code coverage on the FHIR core repo. Currently we have 86% code coverage

- [ ] Add unit tests for broader/maximum code coverage

### PR DESCRIPTION
**Pull request by syedowaisali90**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #326 

**Type**
Choose one: (Bug fix | Feature | Documentation | Testing | Code health | Release | Other) 

**Checklist**
- [x] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [x] I have added any strings visible on UI components to the `strings.xml` file
- [x] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [x] I have built and run the fhircore app to verify my change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/app/src/main/java/org/smartregister/fhircore/activity/BaseSimpleActivity.kt [hunk 1]
```diff
@@ -172,7 +172,7 @@ abstract class BaseSimpleActivity :
     }
   }
 
-  private fun setMenuCounter(@IdRes itemId: Int, count: Int) {
+  fun setMenuCounter(@IdRes itemId: Int, count: Int) {
```
Comment (pld) [inline android/app/src/main/java/org/smartregister/fhircore/activity/BaseSimpleActivity.kt:175]: is this (and visibility changes below) only for testing purpose? If so, I do not think we should do this, here's an article about how to use reflection to test private funs

https://medium.com/mindorks/how-to-unit-test-private-methods-in-java-and-kotlin-d3cae49dccd

File: android/app/src/test/java/org/smartregister/fhircore/activity/PatientListActivityTest.kt [hunk 2]
```diff
@@ -63,6 +67,50 @@ class PatientListActivityTest : ActivityRobolectricTest() {
...
+
+    Assert.assertNotNull(editText)
+    Assert.assertNull(editText.compoundDrawables[0])
+
+    editText.setText("")
+    Assert.assertNotNull(editText.compoundDrawables[0])
+
+    editText.setText("demo")
+    Assert.assertNull(editText.compoundDrawables[0])
+    Assert.assertNotNull(editText.compoundDrawables[2])
```
Comment (pld) [inline android/app/src/test/java/org/smartregister/fhircore/activity/PatientListActivityTest.kt:82]: how do we know to look at index 2?

File: android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+import org.junit.Test
+import org.robolectric.Robolectric
+import org.robolectric.Shadows.shadowOf
+import org.robolectric.annotation.Config
+import org.robolectric.shadows.ShadowAlertDialog
+import org.smartregister.fhircore.R
+import org.smartregister.fhircore.shadow.FhirApplicationShadow
+
+@Config(shadows = [FhirApplicationShadow::class])
+class RecordVaccineActivityTest : ActivityRobolectricTest() {
```
Comment (pld) [inline android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt:38]: great work!

File: android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      }
+
+    recordVaccineActivity =
+      Robolectric.buildActivity(RecordVaccineActivity::class.java, intent).create().resume().get()
+  }
+
+  @Test
+  fun testVerifyRecordedVaccineSavedDialogProperty() {
+
+    // val fragmentManager = mockk<FragmentManager>()
```
Comment (pld) [inline android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt:59]: remove?

**Review by pld — CHANGES_REQUESTED** [review]
looks close! is there a reason this is dnm other than needs review? I think we should merge if not, I'd rather have multiple PRs for multiple tests than a single one pushing it over 80 that way we don't have to worry about it going out of data w/other ongoing code changes

**Comment by syedowaisali90** [comment]
> looks close! is there a reason this is dnm other than needs review? I think we should merge if not, I'd rather have multiple PRs for multiple tests than a single one pushing it over 80 that way we don't have to worry about it going out of data w/other ongoing code changes

Actually, there is no specific reason to add DNM. I was thought I will cover 80% in one PR but we can have multiple PRs for ongoing code changes.

File: android/app/src/test/java/org/smartregister/fhircore/activity/PatientListActivityTest.kt (hunk 2 above)
Comment (syedowaisali90) [reply android/app/src/test/java/org/smartregister/fhircore/activity/PatientListActivityTest.kt:82]: Basically, the order of drawables are fixed. https://github.com/opensrp/fhircore/blob/dfa5467a68d99edf76a6be25ab63b7f361db7530/android/app/src/main/java/org/smartregister/fhircore/activity/PatientListActivity.kt#L63-L68

File: android/app/src/main/java/org/smartregister/fhircore/activity/BaseSimpleActivity.kt (hunk 1 above)
Comment (syedowaisali90) [reply android/app/src/main/java/org/smartregister/fhircore/activity/BaseSimpleActivity.kt:175]: logic updated using with reflection api. Thanks

File: android/app/src/main/java/org/smartregister/fhircore/activity/PatientListActivity.kt [hunk 5]
```diff
@@ -86,7 +86,7 @@ class PatientListActivity : BaseSimpleActivity() {
   }
 
   // pager adapter
-  private inner class PatientListPagerAdapter(fa: FragmentActivity) : FragmentStateAdapter(fa) {
+  inner class PatientListPagerAdapter(fa: FragmentActivity) : FragmentStateAdapter(fa) {
```
Comment (pld) [inline android/app/src/main/java/org/smartregister/fhircore/activity/PatientListActivity.kt:89]: why not keep private?

File: android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt [hunk 6]
```diff
@@ -34,6 +34,7 @@ const val USER_ID = "user_id"
 class RecordVaccineActivity : MultiLanguageBaseActivity() {
 
   private val viewModel: QuestionnaireViewModel by viewModels()
+  lateinit var fragment: QuestionnaireFragment
```
Comment (pld) [inline android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt:37]: why not keep private?

File: android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt (hunk 4 above)
Comment (syedowaisali90) [reply android/app/src/test/java/org/smartregister/fhircore/activity/RecordVaccineActivityTest.kt]: unused code removed.

File: android/app/src/main/java/org/smartregister/fhircore/activity/PatientListActivity.kt (hunk 5 above)
Comment (syedowaisali90) [reply android/app/src/main/java/org/smartregister/fhircore/activity/PatientListActivity.kt]: fixed.

File: android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt (hunk 6 above)
Comment (syedowaisali90) [reply android/app/src/main/java/org/smartregister/fhircore/activity/RecordVaccineActivity.kt]: fixed.

**Review by pld — APPROVED** [review]

**Pull request merged**